ui/app/AppLayouts/Wallet/views/walletconnect/WalletConnectModal.qml

```diff
@@ -230,12 +264,75 @@ Popup {
         function onStatusChanged(message) {
             statusText.text = message
         }
+
+        function onAuthRequest(request) {
+            d.observedData = request
+            d.setStatusText("Select the address you want to sign in with:")
+
+            accountsModel.clear()
+
+            let walletAccounts = root.controller.getWalletAccounts()
+            try {
+                let walletAccountsJsonArr = JSON.parse(walletAccounts)
+
+                for (let i = 0; i < walletAccountsJsonArr.length; i++) {
+                    let obj = {
+                        preferredSharingChainIds: ""
+                    }
+
+                    for (var key in walletAccountsJsonArr[i]) {
+                        obj[key] = walletAccountsJsonArr[i][key]
+                    }
+
+                    accountsModel.append(obj)
+                }
+
+            } catch (e) {
+                console.error("error parsing wallet accounts, error: ", e)
+                d.setStatusText("error parsing walelt accounts", "red")
+                return
+            }
+        }
+
+        function onAuthSignMessage(message, address) {
+            let details = ""
+            if (!!d.observedData.verifyContext.verified.isScam) {
+                details = "This website you`re trying to connect is flagged as malicious by multiple security providers.\nApproving may lead to loss of funds."
+            } else {
+                if (d.observedData.verifyContext.verified.validation === "UNKNOWN")
+                    details = "Website is Unverified"
+                else if (d.observedData.verifyContext.verified.validation === "INVALID")
+                    details = "Website is Mismatched"
+                else
+                    details = "Website is Valid"
+            }
+
+            d.selectedAddress = address
+            d.authMessage = message
+            d.setDetailsText(`${details}\n\n${message}`)
+            d.state = d.waitingUserResponseToAuthRequest
+        }
+
+        function onAuthRequestUserAnswerResult(accept, error) {
+            if (error) {
+                d.setStatusText(`Auth Request ${accept ? "Accept" : "Reject"} error`, "red")
+                return
+            }
+
+            if (accept) {
+                d.setStatusText(`Auth Request completed`)
```

Comment:
Here, we should record the authentication as we do with the pairing where we call `root.controller.recordSuccessfulPairing(JSON.stringify(sessionProposal))`. It could be done in a future PR, though.

Reply:
As discussed in private chat, since we're not sure yet do we need to store anything in the context of auth API, will leave that for later once we discuss that with the UI team and analyze their findings on this topic.